user.js

```diff
@@ -0,0 +1,134 @@
+const environment = process.env.NODE_ENV || 'development';// set environment
+const configuration = require('./knexfile')[environment];// pull in correct db with env configs
+const database = require('knex')(configuration); // define database based on above
+const bcrypt = require('bcrypt') // bcrypt will encrypt passwords to be saved in db
+const crypto = require('crypto') // built-in encryption node module
+
+
+// request object: {
+//        "username": "Alex0@turing.com",
+//        "password": "supersecret7",
+//        "points": 1002
+// }
+
+//POINTS IS NOT WORKING
+const signup = (request, response) => {
+  // get user from request body
+  const user = request.body
+
+  for (let requiredParameter of [
+    'username',
+    'password',
+    ]) {
+    if (user[requiredParameter] === undefined) {
+      response.status(422).send({error: 'Missing required parameter'});
+    } else {
+      encryptPassword(user.password) //promise
+        .then( encryptedPassword => {
+          delete user.password //delete old password
+          user.password_digest = encryptedPassword //assign password_digest to the new encrypted
+        })
+        .then(() => createToken())
+        .then(token => user.token = token)
+        .then(() => createUser(user))
+        .then(user => {
+          delete user.password_digest
+          response.status(201).json({ user })
+        })
+        .catch((error) => console.error(error)) 
+    }
+  }
+}
+  // encrypt plain text password with bcrypt, returns a promise
+  //bcrypt docs : 
+  //  bcrypt.hash(myPlaintextPassword, saltRounds, function(err, hash) {
+  //    Store hash in your password DB.
+  //  });
```

Comment:
Thanks for sending me that lesson so i could figure out what was going on up to this point

Reply:
No Problem!